src/utils.ts

```diff
@@ -132,3 +132,16 @@ export const isDefined = (value: any): boolean => value !== void 0;
 
 export const isMatchingPath = (path: string, criteria: string | RegExp) =>
     isRegex(criteria) ? !!path.match(criteria as RegExp) : checkGlob(path, criteria as string);
+
+export const isExpanded = (path: string, state: JsonViewerState): boolean => state.expanded[path];
+
+export const isCollapsed = (path: string, state: JsonViewerState): boolean => !state.expanded[path];
+
+export const isDeepestVisible = (node: HTMLElement): boolean => node.children[1] instanceof HTMLSpanElement;
+
+export const isImmediateParent = (currentPath: string, compare: string): boolean => currentPath.includes(compare);
```

Comment:
i think for immediate parent this function has to get a lot stricter: if it used `startsWith` it'd be `isAncestor`, actually saying it's an immediate parent is a little tougher i think?

Reply:
Yeah, I guess I don't care if its technically the immediate parent as far as the dom is concerned. I renamed to `isAncestor`

Changes applied